### PR TITLE
Improve debug log for interpreter requests during project commands

### PR DESCRIPTION
**Pull request by zanieb**
While it's slightly more convenient to log this where we were, it was pretty unhelpful e.g.

```
DEBUG Interpreter meets the requested Python: `Python >=3.9`
```

What interpreter are we referring to here?

### CODE REVIEW

File: crates/uv/src/commands/project/mod.rs [hunk 1]
```diff
@@ -148,13 +148,7 @@ fn interpreter_meets_requirements(
...
         return true;
     };
-    if request.satisfied(interpreter, cache) {
-        debug!("Interpreter meets the requested Python: `{request}`");
-        true
-    } else {
-        debug!("Interpreter does not meet the request: `{request}`");
-        false
-    }
+    request.satisfied(interpreter, cache)
```
Comment (charliermarsh) [inline crates/uv/src/commands/project/mod.rs:151]: Honestly should we just get rid of this method? It's `requested_python.map_or(true, |request| request.satisfied(interpreter, cache))`.

File: crates/uv/src/commands/project/mod.rs [hunk 2]
```diff
@@ -231,16 +225,25 @@ impl FoundInterpreter {
                     python_request.as_ref(),
```
Comment (charliermarsh) [inline crates/uv/src/commands/project/mod.rs:225]: Is this used anywhere else?

**Review by charliermarsh — APPROVED** [review]

File: crates/uv/src/commands/project/mod.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv/src/commands/project/mod.rs:151]: I think it used to be used in more places? You're probably right though.

Comment (charliermarsh) [reply crates/uv/src/commands/project/mod.rs:151]: Yeah looks like this is the only one. IMO we should just inline it.

**Pull request merged**